### PR TITLE
DOR-3557: add full path info traversal

### PR DESCRIPTION
**Pull request by adirizka7**

### CODE REVIEW

File: g_docs.go [hunk 1]
```diff
@@ -749,49 +757,56 @@ func (res *objectResource) parse() {
 		ColorLog("Unknown type without TypeSec: %v\n", res.object)
 		os.Exit(1)
 	}
-	// TODO support other types, such as `ArrayType`, `MapType`, `InterfaceType` etc...
```
Comment (fahimbagar) [inline g_docs.go:752]: Are you sure to remove this TODO?

**Review by fahimbagar — COMMENTED** [review]
When I generate docs for DOR, there is a warning:
```
2021/11/22 17:06:09 [WARN]can't find the object: Time
```

Does this mean object `Time` is not supported?

File: g_docs.go [hunk 2]
```diff
@@ -822,7 +837,7 @@ func objectWithPackageName(object, packageName string) string {
 // constructObjectPropertie constructs a swagger.Propertie out of
 // an ast.Expr. This function recursively traverse all expression
 // until it reaches one of object or pre-defined basic golang type / primitive.
-func constructObjectPropertie(field ast.Expr, packageName string, realTypes *[]string) swagger.Propertie {
+func constructObjectPropertie(field ast.Expr, packageName string, realTypes *[]string, pathInfo map[string]string) swagger.Propertie {
 	var propertie swagger.Propertie
```
[the remark this answers is not in the document]
Comment (adirizka7) [reply g_docs.go:841]: yeah it also feels weird to me 😅, I wanted to make it `property` the other day but ended up just followed the `swagger` own name for the sake of consistency with the native package name 😆

File: g_docs.go (hunk 1 above)
Comment (adirizka7) [reply g_docs.go:752]: oh good catch! These types are currently not used in DOR. I wanted to add a log in the `default:` section of the switch case, so we can address it right away when we really need to use these types. 

Will add some logs in the `default:`

**Comment by adirizka7** [comment]
> When I generate docs for DOR, there is a warning:
> 2021/11/22 17:06:09 [WARN]can't find the object: Time
> Does this mean object Time is not supported?

I tried it in my local, and can't find similar warning
```
➜  doraemon git:(develop) bee generate docs
______
| ___ \
| |_/ /  ___   ___
| ___ \ / _ \ / _ \
| |_/ /|  __/|  __/
\____/  \___| \___| v1.5.2
2021/11/22 17:18:24 [WARN] missing description from '200' Response for route DELETE '/customers/'
2021/11/22 17:18:24 [SUCC] Docs successfully generated!
```
which branch were you using when running this? 🤔

File: g_docs.go (hunk 1 above)
Comment (adirizka7) [reply g_docs.go:752]: Addressed in https://github.com/zalora/bee/pull/18/commits/8d8c52a78db734e78368178ee617124bd85988a3

**Comment by fahimbagar** [comment]
> > When I generate docs for DOR, there is a warning:
> > 2021/11/22 17:06:09 [WARN]can't find the object: Time
> > Does this mean object Time is not supported?
> 
> I tried it in my local, and can't find similar warning
> 
> ```
> ➜  doraemon git:(develop) bee generate docs
> ______
> | ___ \
> | |_/ /  ___   ___
> | ___ \ / _ \ / _ \
> | |_/ /|  __/|  __/
> \____/  \___| \___| v1.5.2
> 2021/11/22 17:18:24 [WARN] missing description from '200' Response for route DELETE '/customers/'
> 2021/11/22 17:18:24 [SUCC] Docs successfully generated!
> ```
> 
> which branch were you using when running this? thinking

You could use one of these brances:
- DOR-3561
- DOR-3561-staging (rebased from current staging branch)
- staging

Let me know if this is not an issue.

**Comment by adirizka7** [comment]
> You could use one of these brances:
> - DOR-3561
> - DOR-3561-staging (rebased from current staging branch)
> - staging
> Let me know if this is not an issue.

Nice finding @fahimbagar ! It was indeed unsupported, yet. Addressed it in https://github.com/zalora/bee/pull/18/commits/14edb2c725bf0cbd7ecde4468dad12d60980789e . could you try again in those branches from your side?

**Comment by fahimbagar** [comment]
> 

awesome, it works now

**Pull request merged**